actor/builtin/paymentbroker/paymentbroker_test.go

```diff
@@ -195,6 +195,52 @@ func TestPaymentBrokerRedeemWithCondition(t *testing.T) {
 	})
 }
 
+func TestPaymentBrokerRedeemReversesCancellations(t *testing.T) {
+	tf.UnitTest(t)
+
+	t.Run("resets EOL on success", func(t *testing.T) {
```

Comment:
This test is correct, but requires a lot of internal knowledge to understand why. These tests should serve as documentation. Please comment the purpose of the steps, and add an intermediate assertion after the cancel for `AgreedEol` and `Eol` to demonstrate that redeem is actually resetting those values.